### PR TITLE
[TRAVEL] Results filter

### PR DESCRIPTION
**Pull request by gifteconomist**
Adding the filter component for the Traveler template results page. Link to component in staging:

https://travel-results-filter.firebaseapp.com/templates/travel/travel-results.amp

All travel-result-page components use a [src] bound amp-list element to pass attributes to filter the search results by. This filter component displays city, departure/return, max price, sorting and activity-type parameters, and allows you to alter activity-type, city, max-price, and sorting-parameters to alter the query. This component filters which results are displayed when paired with the "results" component.

### CODE REVIEW

**Comment by camelburrito** [comment]
@gifteconomist this PR is not rebased right - please sync with master :) , it shows 5000+ lines of code !!

File: css/templates/travel/components/_city-selector.css [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  transition: border-color 300ms;
+}
+
+.travel-city-selector [option]:focus::before {
+  border-color: #a480e3;
+}
+
+.travel-city-selector [option]::after {
+  content: "";
```
Comment (camelburrito) [inline css/templates/travel/components/_city-selector.css]: nit: single quotes

File: css/templates/travel/components/_city-selector.css [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  transition: transform 300ms;
+}
+
+.travel-city-selector [option]:hover .travel-city-selector-img,
+.travel-city-selector [option]:focus .travel-city-selector-img {
+  transform: scale(1.2);
+}
+
+/* Need to target amp-img directly to prevent strange bug where amp-img elements randomly lose rounded corners. */
+amp-img.rounded img {
```
Comment (camelburrito) [inline css/templates/travel/components/_city-selector.css]: these rules targeting a general amp element should move to the main page.css

File: css/templates/travel/components/_filter-bar.css [hunk 3]
```diff
@@ -0,0 +1,16 @@
+.travel-filter-bar {
+  animation: fade-and-slideup 0.4s 0.05s ease-in-out both;
+}
+
+.travel-filter-bar-collapse {
+  height: 0;
+  overflow: hidden;
+}
+
+.travel-filter-bar .gray {
```
Comment (camelburrito) [inline css/templates/travel/components/_filter-bar.css]: would prefer not to use the bass color classes  let me know what you think. Also i would definitely stay away from this type of usage. just declare another class to override

File: css/templates/travel/components/_type-selector.css [hunk 4]
```diff
@@ -0,0 +1,113 @@
+/**
```
Comment (camelburrito) [inline css/templates/travel/components/_type-selector.css]: copyrights on all files

File: css/templates/travel/components/_city-selector.css [hunk 5]
```diff
@@ -0,0 +1,117 @@
+/**
```
Comment (camelburrito) [inline css/templates/travel/components/_city-selector.css]: copyrights on all files

File: css/templates/travel/components/_type-selector.css [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  border-color: rgba(0,0,0,0.25);
+}
+
+.travel-type-selector-active {
+  transition-delay: 0.3s;
+}
+
+.travel-type-selector-inactive::after,
+.travel-type-selector-active::after {
+  content: "+";
```
Comment (camelburrito) [inline css/templates/travel/components/_type-selector.css]: nit: single quotes

File: css/templates/travel/elements/_border.css [hunk 7]
```diff
@@ -0,0 +1,8 @@
+/*
+ * border
+ *
+ */
+
+ .travel-border-gray {
```
Comment (camelburrito) [inline css/templates/travel/elements/_border.css]: dont create files for one off classes. also fix the name not a good name

Comment (camelburrito) [reply css/templates/travel/elements/_border.css]: if you want to change the border color for this template there is already a variable to override

File: css/templates/travel/elements/_border.css [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+   border-color: var(--silver);
```
Comment (camelburrito) [inline css/templates/travel/elements/_border.css]: rename color variables to prefix --color

File: css/templates/travel/elements/_fade-right.css [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+ * Create an element that fades out overflow content to the right.
+ */
+.travel-fade-right {
+  overflow: hidden;
+  white-space: nowrap;
+  position: relative;
+}
+
+.travel-fade-right::after {
+  content: "";
```
Comment (camelburrito) [inline css/templates/travel/elements/_fade-right.css]: nit: ''

File: css/templates/travel/elements/_input-range.css [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+ * input-range
+ *
+ * Create a stylized input[type=range] element.
+ */
+.travel-input-range {
+  display: block;
+  background-color: transparent;
+  color: inherit;
+  background-color: transparent;
+  -webkit-appearance: none;
```
Comment (camelburrito) [inline css/templates/travel/elements/_input-range.css]: make the properties with - the last set of properties

File: css/templates/travel/elements/_input-range.css [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+.travel-input-range::-webkit-slider-thumb {
+  position: relative;
+  width: var(--range-thumb-size);
+  height: var(--range-thumb-size);
+  cursor: pointer;
+  margin-top: calc((var(--range-track-height) - var(--range-thumb-size)) * 0.5);
+  border-radius: calc(var(--range-thumb-size) * 0.5);
+  background-color: var(--range-thumb-color);
+  -webkit-appearance: none;
```
Comment (camelburrito) [inline css/templates/travel/elements/_input-range.css]: move down

File: css/templates/travel/elements/_list-selector.css [hunk 12]
```diff
@@ -0,0 +1,24 @@
+
+/**
+ * list-selector
+ *
+ * Style an <amp-selector> element
+ */
+.travel-list-selector [option] {
+  padding: 6px 12px;
+  border-radius: var(--border-radius);
+  background: var(--white);
```
Comment (camelburrito) [inline css/templates/travel/elements/_list-selector.css]: fix color

File: css/templates/travel/elements/_overlay.css [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+ * at different breakpoints.
+ *
+ * The required markup is:
+ *  div.overlay(.overlay-visible) > div.overlay-overflow > div.overlay-content
+ *
+ * Modifier Classes:
+ * - overlay-visible
+ */
+
+.travel-overlay {
```
Comment (camelburrito) [inline css/templates/travel/elements/_overlay.css]: I think a lot of the properties can be moved to html?

File: templates/travel/partials/filters/filters.snip.html [hunk 14]
```diff
@@ -0,0 +1,32 @@
+<!-- Filters -->
+<div
+  class="travel-filters clearfix mxn2"
+  [class]="'travel-filters clearfix mxn2' + (ui_reset ? ' travel-filters-reset' : '')"
+>
+  <div class="col col-12 md-col-4 px2 py1">
+    <div class="h5 gray mb1">MAX PRICE</div>
```
Comment (camelburrito) [inline templates/travel/partials/filters/filters.snip.html]: should you be doing caps (class) instead?

Comment (camelburrito) [inline templates/travel/partials/filters/filters.snip.html]: This entire thing should be a definition list 
```html
<dl>
</dl>
```
with dts and dds

File: templates/travel/partials/filters/price-range.snip.html [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+  see: https://github.com/ampproject/amphtml/issues/8214
+-->
+<svg viewBox="0 0 800 100" class="travel-input-range-shape col-12 block">
+  <defs>
+    <linearGradient x1="0" y1="0" x2="0" y2="120%" id="price-gradient">
+      <stop offset="0%" stop-color="currentColor"></stop>
+      <stop offset="200%" stop-color="#FFF"></stop>
+    </linearGradient>
+  </defs>
+  <path fill="url(#price-gradient)" d="m0,100 S12.5 100,25 100 S62.5 91.477,75 82.955 S112.5 64.204,125 62.5 S162.5 55.966,175 52.841 S212.5 41.477,225 36.364 S262.5 28.125,275 30.113 S312.5 43.466,325 52.841 S362.5 76.421,375 81.25 S412.5 85.227,425 79.546 S462.5 51.137,475 34.091 S512.5 1.421,525 2.841 S562.5 15.057,575 24.432 S612.5 45.881,625 48.579 S662.5 56.108,675 58.239 S712.5 71.875,725 81.25 S762.5 100,775 100 S800 100,800 100"></path>
```
Comment (camelburrito) [inline templates/travel/partials/filters/price-range.snip.html]: is this optmized with svgo?

File: css/templates/travel/elements/_overlay.css (hunk 13 above)
Comment (gifteconomist) [reply css/templates/travel/elements/_overlay.css]: done

File: css/templates/travel/elements/_input-range.css (hunk 10 above)
Comment (gifteconomist) [reply css/templates/travel/elements/_input-range.css]: done

File: css/templates/travel/elements/_fade-right.css (hunk 9 above)
Comment (gifteconomist) [reply css/templates/travel/elements/_fade-right.css]: done

File: css/templates/travel/components/_type-selector.css (hunk 6 above)
Comment (gifteconomist) [reply css/templates/travel/components/_type-selector.css]: done

File: css/templates/travel/components/_type-selector.css (hunk 4 above)
Comment (gifteconomist) [reply css/templates/travel/components/_type-selector.css]: done

File: css/templates/travel/components/_city-selector.css (hunk 5 above)
Comment (gifteconomist) [reply css/templates/travel/components/_city-selector.css]: done

**Comment by gifteconomist** [comment]
@camelburrito I believe everything is addressed, sans changing variable names for colors (waiting until merge so that this review remains specific to the filter component).

File: css/templates/travel/components/_city-selector.css [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+  /* Right offset is equal to the focus shadow width. */
+  right: 3px;
+  top: 0;
+  bottom: 0;
+  margin: auto;
+}
+
+
+.travel-city-selector [option]::before {
+  content: "";
```
Comment (camelburrito) [inline css/templates/travel/components/_city-selector.css]: ''

File: css/templates/travel/components/_filter-overlay.css [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ .travel-overlay-header, .travel-overlay-footer {
+  background-color: var(--white);
```
Comment (camelburrito) [inline css/templates/travel/components/_filter-overlay.css]: --color-white

File: css/templates/travel/components/_filter-overlay.css [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+.travel-overlay-reset-btn {
+  cursor: pointer;
+  color: var(--gray);
```
Comment (camelburrito) [inline css/templates/travel/components/_filter-overlay.css]: --color-gray

File: css/templates/travel/components/_filters.css [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+
+.travel-filters-reset {
+  animation: travel-filters-reset 0.82s cubic-bezier(.36,.07,.19,.97) both;
+  transform: translate3d(0, 0, 0);
+  backface-visibility: hidden;
+  perspective: 1000px;
+}
+
+.travel-filters-apply-btn {
+  background-color: var(--purple);
```
Comment (camelburrito) [inline css/templates/travel/components/_filters.css]: --color-purple

File: css/templates/travel/elements/_inline-list.css [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+/*
+  Add an nbsp character to the inline-list element to ensure prevent it
+  from shrinking to 0px X 0px when otherwise empty which would make AMP think
+  the element was not visible in the viewport and would prevent the list
+  request from firing.
+ */
+.travel-inline-list::after {
+  content: "\00a0";
```
Comment (camelburrito) [inline css/templates/travel/elements/_inline-list.css]: i think `content: ' '` would work

File: css/templates/travel/elements/_input-range.css (hunk 10 above)
Comment (camelburrito) [reply css/templates/travel/elements/_input-range.css]: the property is still not the last property?

File: css/templates/travel/components/_filters.css (hunk 19 above)
Comment (camelburrito) [reply css/templates/travel/components/_filters.css]: plus below

File: css/templates/travel/components/_type-selector.css [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+  position: relative;
+}
+
+.travel-type-selector-inactive,
+.travel-type-selector-active {
+  height: 38px;
+  padding-left: 12px;
+  padding-right: 10px;
+  border-radius: 19px;
+  border: 2px solid var(--silver);
```
Comment (camelburrito) [inline css/templates/travel/components/_type-selector.css]: --color-silver

File: css/templates/travel/elements/_badge.css [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+ /*
+ * badge
+ *
+ * Add a background and border to an inline element.
+ */
+.travel-badge {
+  padding: 4px 10px;
+  color: var(--gray);
```
Comment (camelburrito) [inline css/templates/travel/elements/_badge.css]: --color prefixes please

File: css/templates/travel/components/_city-selector.css (hunk 16 above)
Comment (bentranter) [reply css/templates/travel/components/_city-selector.css]: Sorry, I'm not sure what you mean by this @camelburrito

Comment (bentranter) [reply css/templates/travel/components/_city-selector.css]: Never mind, I understand

File: css/templates/travel/elements/_pane.css [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+  pointer-events: none;
+}
+
+.travel-pane-overflow {
+  opacity: 0;
+  transition: opacity 100ms;
+}
+
+.travel-pane-content {
+  background-color: var(--white);
```
Comment (camelburrito) [inline css/templates/travel/elements/_pane.css]: --color-white?

File: css/templates/travel/elements/_input-range.css [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+.travel-input-range:focus {
+  outline: none;
+}
+
+.travel-input-range::-moz-focus-outer {
+  border: 0;
+}
+
+.travel-input-range-shape {
+  color: var(--lime);
```
Comment (camelburrito) [inline css/templates/travel/elements/_input-range.css]: --color-?

File: css/templates/travel/elements/_row-selector.css [hunk 25]
```diff
@@ -0,0 +1,61 @@
...
+ */
+.travel-row-selector {
+  overflow-y: hidden;
+  overflow-x: auto;
+  white-space: nowrap;
+}
+
+.travel-row-selector [option] {
+  display: inline-block;
+  color: var(--gray);
```
Comment (camelburrito) [inline css/templates/travel/elements/_row-selector.css]: --color-

File: css/templates/travel/elements/_row-selector.css [hunk 26]
```diff
@@ -0,0 +1,61 @@
...
+  padding-top: 1rem;
+}
+
+.travel-row-selector [option]:not(:last-of-type) {
+  margin-right: 2rem;
+}
+
+.travel-row-selector [option][selected] {
+  color: var(--green);
```
Comment (camelburrito) [inline css/templates/travel/elements/_row-selector.css]: --color-

**Comment by bentranter** [comment]
I Signed it

**Pull request closed without merging**